### PR TITLE
Updated pdf version specs

### PR DESCRIPTION
**Pull request by Arshitha**
Bugs Fixed: 
- Stable Table of Contents page
- Cover page 
- Non-tabular text overflow 
- Overlapping columns of tables (owing to small widths of the pipe tables in the markdown files)
  - Markdown files were edited (one-time edit) to increase the column widths of pipe tables
  - Not sure how this would affect the read the docs version of specs
- Automating header to include the latest release version number and date 

Issues that need to be fixed: 
- Internal links referencing either the same markdown file or other markdown files break during conversion from markdown to pdf. Currently, all of the internal links are removed and replaced by the accompanying text. 
- Superscripts are lost in conversion as well 
- Tried a couple different family of fonts to fix the missing character warnings for emojis used in the markdowns with no success. It would be ideal to fix this to match the Read The Docs version more closely
- Entity table in Appendix IV has overlapping unreadable columns since it's markdown equivalent has more than 10 columns. Refer to issue for more discussion

### CODE REVIEW

File: src/CHANGES.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-Changelog
+# Changelog
```
Comment (sappelhoff) [inline src/CHANGES.md]: interestingly, it seems like the rendered changelog page gets a "changelog" heading inserted automatically --> https://bids-specification.readthedocs.io/en/stable/CHANGES.html

previously we then had something like

---

# Changelog

Changelog

---

is it possible that this change will turn this into

---

# Changelog

# Changelog

---

?

**Comment by nicholst** [comment]
Lovely!

The PR comment suggests that *no* internal links work... that's not strictly true: The Table of Contents links works, which is great.

Line wrapping of code is working great now.

Very long words remain a challenge, e.g. on pg 39 see `DigitizedHeadPointsCoordinateSystemDescription` and the period that falls on the next line, but, really, if that's the worst of it I'm very happy.

Footnotes render particularly oddly... I must admit I didn't realise we allow footnotes... in a HTML-centric view of the world, footnotes are a problem, and parenthesis would seem to be preferred, but at least here in a paginated PDF-world you'd hope they would appear as footnotes.   (I guess I should start a new PR suggesting we ban footnotes, but for now...)  ANYWAY, ideally footnotes would appear as footers, but it's not really a big deal... but they *should* render like they do in HTML -- i.e. as superscripted.

**Comment by nicholst** [comment]
Sorry, I forgot to add one more comment/enhancement:  The title page should have more than the BIDS logo... it should have the title and the version.

As inspiration, consider this [IEEE Standard](https://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&arnumber=8653544)... it has the name of the document and the authors and the version/revision number.

File: src/CHANGES.md (hunk 1 above)
Comment (Arshitha) [reply src/CHANGES.md]: @sappelhoff Thanks for the feedback!

- I've reverted the change I'd made. So I believe that should have fixed it. I needed "Changelog" to be Heading 1 so that pandoc recognizes it to be a new section and doesn't concatenate with the previous section. However, I found a hack around it that didn't require the 'src/CHANGES.md' to be modified. 

- Most recent commit takes care of naming change and doesn't duplicate the pdf. I'm looking into how to make it part of the build and render it as part of the RTD

**Comment by Arshitha** [comment]
@nicholst Thanks for both the comments. I'll update the PR soon.

**Comment by Arshitha** [comment]
sorry, I'm traveling internationally over the last couple of weeks with spotty internet, so couldn't respond to the comments earlier.

File: src/CHANGES.md (hunk 1 above)
Comment (sappelhoff) [reply src/CHANGES.md]: > I've reverted the change I'd made. So I believe that should have fixed it. I needed "Changelog" to be Heading 1 so that pandoc recognizes it to be a new section and doesn't concatenate with the previous section. However, I found a hack around it that didn't require the 'src/CHANGES.md' to be modified.

actually ... perhaps it was good that you turned that "changelog" into a heading. Perhaps that would make it findable on our website: https://bids-specification.readthedocs.io/en/stable/CHANGES.html

when typing "changelog" into the search function, I don't get this site ... although the search function usually qorks quite well. :thinking: 

@franklin-feingold would it be a problem with our auto-changelog generator to change the first line in https://github.com/bids-standard/bids-specification/blob/master/src/CHANGES.md to be a markdown heading?

Comment (franklin-feingold) [reply src/CHANGES.md]: if we change it the changelog generator regenerates (and commits) for each merged PR so that won't solve it - there may be something within the changelog generator that can be changed but nothing apparent

**Comment by franklin-feingold** [comment]
Thank you @Arshitha for your time and effort putting this together - looks great!! 

+100 to both @sappelhoff and @nicholst comments for reaching the viable release, perhaps iterating after these comments are addressed? then can start a new issue to track further enhancements to the pdf rendering

File: src/CHANGES.md (hunk 1 above)
Comment (Arshitha) [reply src/CHANGES.md]: So do I leave it as it is right now? or make the "changelog" a heading?

**Comment by Arshitha** [comment]
@nicholst I've added title and version details (pulling from changelog and so, automated) to the cover page. However, it's not as aesthetically pleasing as the [IEEE Standard](https://ieeexplore.ieee.org/document/8653544) you'd referenced. Perhaps, this could be something that can be improved as part of a later issue?

File: src/CHANGES.md (hunk 1 above)
Comment (sappelhoff) [reply src/CHANGES.md]: We leave it as it is for now and look into whether we can tweak some changelog-generator settings to label the page with `# changelog` instead of `changelog`

Comment (sappelhoff) [reply src/CHANGES.md]: we'll see if https://github.com/bids-standard/bids-specification/pull/417 does the job.

Comment (Arshitha) [reply src/CHANGES.md]: sounds good!

**Comment by nicholst** [comment]
Sorry... where can I see the current PDF associated with this PR?

**Comment by Arshitha** [comment]
Apologies, it was on my forked version of the repo. Here is the link: https://github.com/Arshitha/bids-specification/tree/updated-pdf-version-specs/pdf_build_src
But here is a copy just for easy reference: 
[bids-spec.pdf](https://github.com/bids-standard/bids-specification/files/4220923/bids-spec.pdf)

**Comment by nicholst** [comment]
The cover page looks great!

Other thoughts/feedback:  Are the "Chapter ?" labels inevitable?  As a PDF, they don't bother me at all, but that they are missing from the RTD rendering is annoying.

ACTUALLY, I wonder if the RTD version would be better off to have the numbering!  Most official standards have numbered sections, to make it easier to refer to different portions of a standard.  Though I guess that's another issue.

ANOTHER weirdness that may be not so much about the PDF:  The chapter titles don't match the headings!   What is Chapter 13 titled "BIDS Extension Proposals", on RTD has a left-column label of "Extending the BIDS specification"... why the mismatch there?  Does the "Extending the BIDS specification" text ever appear in the PDF?  Again, seems not so much a problem for this PDF rendering as a structural issue.

I flipped through and caught a few other things:

In Chapter 10 "Physiological and other continuous recordings", the last sentence of the first parargraph ends with a hyperlink, but there's nothing to link, i.e. 
>  For example:%60task-movie_stim.tsv.gz‘
which is rendered without a problem in the RTD version.

Also, really minor... Chapter 14 lacks emojis.  Emoji's seem challenging in LaTeX but there is some style that does it: https://github.com/alecjacobson/coloremoji.sty

**Comment by Arshitha** [comment]
@nicholst Thanks for that detailed feedback! 

> Are the "Chapter ?" labels inevitable? As a PDF, they don't bother me at all, but that they are missing from the RTD rendering is annoying.

Separating sections as chapters is an option provided by pandoc and therefore, most stable solution. There are other hacks but none (that I could find) as stable. I've discussed it in a little more detail here: https://github.com/bids-standard/bids-specification/issues/135#issuecomment-576447907


> ANOTHER weirdness that may be not so much about the PDF: The chapter titles don't match the headings! What is Chapter 13 titled "BIDS Extension Proposals", on RTD has a left-column label of "Extending the BIDS specification"... why the mismatch there? Does the "Extending the BIDS specification" text ever appear in the PDF? Again, seems not so much a problem for this PDF rendering as a structural issue.

This seems to be case because the `mkdocs.yml` file here: https://github.com/bids-standard/bids-specification/blob/dbdf32e7790eb7155bdbc57a8ff93f9ba2b9ebde/mkdocs.yml 
on line 21 has `- Extending the BIDS specification: 06-extensions.md` which is the primary config file used to build the READTHEDOCS version. The pdf chapters are defined by level 1 headings so, perhaps, changing the `mkdocs.yml` would be better? 

> In Chapter 10 "Physiological and other continuous recordings", the last sentence of the first parargraph ends with a hyperlink, but there's nothing to link, i.e.

There was a spacing error in the corresponding .md file. Fixed it now

> Also, really minor... Chapter 14 lacks emojis. Emoji's seem challenging in LaTeX but there is some style that does it: https://github.com/alecjacobson/coloremoji.sty

Thanks a lot for sharing that resource! I gave it quick shot but it required additional packages. I'll add this as part of a separate enhancement issue and take it from there.

File: src/CHANGES.md (hunk 1 above)
Comment (Arshitha) [reply src/CHANGES.md]: @sappelhoff I spent some time looking up how this can be done 
> the pdf should be built concurrently with the READTHEDOCS build of the website, and then also hosted as a build artifact on readthedocs.
According to RTD's build process using `mkdocs`, simultaneous pdf generation isn't possible. https://docs.readthedocs.io/en/stable/builds.html
Perhaps, pdf can be created and then hosted on the built RTD version? However, I'm unsure how to fix this. Would appreciate any ideas/suggestions.

**Comment by nicholst** [comment]
@Arshitha: So, the only question I have is whether the `mkdocs.yml` edits to align 'chapter' and navigation titles can be done as part of this PR, or whether that is 'substantive' enough that it deserves a separate PR.  Thoughts @sappelhoff ?  

Moreover, if they are to be aligned can it be 'institutionalised' somehow, e.g. can titles in markdown be pulled from the YAML or vice-versa?  (Would seem to be a huge pain, on reflection).

File: src/CHANGES.md (hunk 1 above)
Comment (sappelhoff) [reply src/CHANGES.md]: I discussed with @franklin-feingold and @rwblair yesterday and we agreed that the most pragmatic way to go forward would be to split our PDF-keeping in two parts:

1. on each merged pull request to bids-specification, circle CI should build a pdf ... including the current version and date ... this pdf is then hosted in the "artifacts" tab of the circleCI build. --> we can generate a link that will always resolve to the most recently build pdf artifact on circleci. See this link as an example from the mne-bids repo: `https://circleci.com/api/v1.1/project/github/mne-tools/mne-bids/latest/artifacts/0/html/index.html?branch=master` ... it will always resolve to the **latest** index.html file built by circleci

2. whenever we do a **Release** of the bids specification, a maintainer builds the PDF and goes through the steps to host it permanently somewhere ... as currently discussed in https://github.com/bids-standard/bids-specification/pull/407

so for now, we simply need to figure out how to build the pdf using your script ... in circleci. Which is mainly a question of installing all needed packages there.

Comment (sappelhoff) [reply src/CHANGES.md]: this may come in very handy for getting latex to run on circleci: https://github.com/dante-ev/docker-texlive

File: src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md [hunk 2]
```diff
@@ -52,7 +52,7 @@ by Ben Inglis:
 | TotalReadoutTime               | RECOMMENDED. This is actually the "effective" total readout time , defined as the readout duration, specified in seconds, that would have generated data with the given level of distortion. It is NOT the actual, physical duration of the readout train. If `EffectiveEchoSpacing` has been properly computed, it is just `EffectiveEchoSpacing * (ReconMatrixPE - 1)`.<sup>3</sup> . This parameter is REQUIRED if corresponding "field/distortion" maps acquired with opposing phase encoding directions are present (see 8.9.4).                                                                                                                                                                                                                                                                                                  |
 
 <sup>2</sup>Conveniently, for Siemens’ data, this value is easily obtained as
-1/\[`BWPPPE` \* `ReconMatrixPE`\], where BWPPPE is the
+1/\[`BWPPPE` \* `ReconMatrixPE`], where BWPPPE is the
```
Comment (sappelhoff) [inline src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md]: Does the `]` still render correctly after removing the escape character? If yes, why not remove the other escape character right before `[`BWPPPE`` as well?

File: src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md [hunk 3]
```diff
@@ -64,39 +64,40 @@ first "effective" echo and the center of the last "effective" echo.
...
+| DwellTime              | RECOMMENDED. Actual dwell time (in seconds) of the receiver per point in the readout direction, including any oversampling. For Siemens, this corresponds to DICOM field (0019,1018) (in ns). This value is necessary for the optional readout distortion correction of anatomicals in the HCP Pipelines. It also usefully provides a handle on the readout bandwidth, which isn’t captured in the other metadata tags. Not to be confused with `EffectiveEchoSpacing`, and the frequent mislabeling of echo spacing (which is spacing in the phase encoding direction) as "dwell time" (which is spacing in the readout direction).                                                                                                                                                                                                                                                                                                                                               |
```
Comment (sappelhoff) [inline src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md]: the huge diff across this table is badly rendered unfortunately. I assume you only changed the table borders again?

File: src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md [hunk 4]
```diff
@@ -119,22 +120,22 @@ sub-<label>/[ses-<label>/]
...
+| Name               | `modality_label` | Description                                                                                                                                       |
+| ------------------------- | ------------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------- |
+| T1 weighted        | T1w              |                                                                                                                                                   |
+| T2 weighted        | T2w              |                                                                                                                                                   |
+| T1 Rho map         | T1rho            | Quantitative T1rho brain imaging <br> <https://www.ncbi.nlm.nih.gov/pubmed/24474423> <br> <https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4346383/> |
```
Comment (sappelhoff) [inline src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md]: Why did you change the rendering of these links? If the links break the pandoc process, then I would find it acceptable to put the links as simple text in there 

... what I don't get are your added `<` and `>` characters

File: src/99-appendices/07-meg-systems.md [hunk 5]
```diff
@@ -8,13 +8,13 @@ Restricted keywords for Manufacturer field in the `*_meg.json` file:
 -   [`KIT/Yokogawa/Ricoh`](06-meg-file-formats.md#kityokogawaricoh)
 -   [`KRISS`](06-meg-file-formats.md#kriss)
 -   [`ITAB`](06-meg-file-formats.md#itab)
--   [`Aalto/MEG–MRI`](06-meg-file-formats.md#aalto-megmri)
+-   [`Aalto/MEG/MRI`](06-meg-file-formats.md#aalto-megmri)
```
Comment (sappelhoff) [inline src/99-appendices/07-meg-systems.md]: Why did you perform this change? I am afraid we cannot change this restricted keyword.

File: src/99-appendices/07-meg-systems.md [hunk 6]
```diff
@@ -27,4 +27,4 @@ Restricted keywords for ManufacturersModelName field in the `*_meg.json` file:
 | KIT-160               | KIT/Yokogawa          |                                                                                              |
 | KIT-208               | KIT/Yokogawa          |                                                                                              |
 | ITAB-ARGOS153         | ITAB                  |                                                                                              |
-| Aalto-MEG-MRI-YYYY/MM | Aalto/MEG–MRI         | YYYY-MM (year, month; or major version)                                                      |
+| Aalto-MEG-MRI-YYYY/MM | Aalto/MEG/MRI         | YYYY-MM (year, month; or major version)                                                      |
```
Comment (sappelhoff) [inline src/99-appendices/07-meg-systems.md]: same here: I think we cannot change this restricted keyword

File: pdf_build_src/cover.tex [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+%----------------------------------------------------------------------------------------
+
+\HRule \\[0.4cm]
+{ \huge \bfseries Brain Imaging Data Structure Specification}\\[0.4cm] % Title of your document
+\HRule \\[1.5cm]
+
+% \textsc{\large v1.2.1}\\[0.5cm]{\large 2019-08-14}\\[2cm]
+
+% \vfill % Fill the rest of the page with whitespace
+\textsc{\large v1.2.2}\\[0.5cm]{\large 2020-02-12}\\[2cm]\vfill\end{titlepage}
```
[the remark this answers is not in the document]
Comment (sappelhoff) [reply pdf_build_src/cover.tex]: EDIT: I realize now that you DO automatically extract version and date within your `pdf_build_src/process_markdowns.py` file.

However, please extract the version from the `mkdocs.yml`, because that's the file we mainly update for each release, as described in [the protocol](https://github.com/bids-standard/bids-specification/blob/master/Release_Protocol.md)

Also: I would highly recommend to "inject" the following line into the `cover.tex` file **during the build process**:

`\textsc{\large v1.2.2}\\[0.5cm]{\large 2020-02-12}\\[2cm]\vfill\end{titlepage}`

instead of having it already there and then merely update it during each build.

The advantage of this is, that for each release, `git` will not have to track changes of `cover.tex` ... and the functionality would be the same.

Also, it'd be less confusing to reviewers. 

In short: remove line `27` from `cover.tex` and instead *format* this line in `process_markdowns.py` and also write it to `cover.tex` FROM `process_markdowns.py` **during the build process**

Do similarly for `header.tex`

**Comment by sappelhoff** [comment]
> @Arshitha: So, the only question I have is whether the mkdocs.yml edits to align 'chapter' and navigation titles can be done as part of this PR, or whether that is 'substantive' enough that it deserves a separate PR. Thoughts @sappelhoff ?

> Moreover, if they are to be aligned can it be 'institutionalised' somehow, e.g. can titles in markdown be pulled from the YAML or vice-versa? (Would seem to be a huge pain, on reflection).

@nicholst I think this would be worth discussing in a separate issue.

In my current review I tried to outline only the "absolutely necessary and required changes before a merge" ... so that we can go ahead with a merge and then divide and conquer the remaining points!

**Comment by nicholst** [comment]
This sounds fine to me @sappelhoff... I’ll start a separate issue, maybe PR
if I have  time. --
_________________________________________________________
Thomas Nichols, PhD
Professor of Neuroimaging Statistics
Nuffield Department of Population Health | University of Oxford
Big Data Institute | Li Ka Shing Centre for Health Information and Discovery
Old Road Campus | Headington | Oxford | OX3 7LF | United Kingdom
T: +44 1865 743590 | E: thomas.nichols@bdi.ox.ac.uk
W: http://nisox.org | http://www.bdi.ox.ac.uk

File: src/99-appendices/07-meg-systems.md (hunk 6 above)
Comment (Arshitha) [reply src/99-appendices/07-meg-systems.md]: I'd made this change because for some reason LaTeX to pdf was  errori-ng out with hyphens. I don't recall why that was happening but I've reverted it back to `Aalto/MEG–MRI` on line 30 and line 11. Tested the pdf build and it's working fine. Will make a commit once I address the remaining issues.

File: src/99-appendices/07-meg-systems.md (hunk 5 above)
Comment (Arshitha) [reply src/99-appendices/07-meg-systems.md]: resolved here https://github.com/bids-standard/bids-specification/pull/400/files/1ea328ea38dc07811c4139c70bec13a5a2feb93e#r381360490

File: src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md (hunk 2 above)
Comment (Arshitha) [reply src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md]: I don't remember making this change, it was accidental, I think. Renders fine with or without escape characters but have reverted it back to include escape character to be consistent. Thanks for pointing this out.

File: src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md (hunk 3 above)
Comment (Arshitha) [reply src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md]: Yes, I only changed line 68. Do you mean the table is badly rendered on RTD version or the pdf?

Comment (sappelhoff) [reply src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md]: No I just meant that I did not check in detail, because the red/green markup didn't show properly what was changed and what was not changed :-)

File: pdf_build_src/build_pdf.sh [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+cd src_copy
+python pandoc_script.py
+mv bids-spec.pdf ..
+cd ..
+
+# delete the duplicated src directory
+rm -rf src_copy
+
+# open bids-specs.pdf
+open bids-spec.pdf
```
[the remark this answers is not in the document]
Comment (Arshitha) [reply pdf_build_src/build_pdf.sh]: Yes, it was for convenience. Removed it now.

File: .circleci/config.yml [hunk 9]
```diff
@@ -29,6 +29,16 @@ jobs:
...
 
+  build_docs_pdf:
+    docker:
+      - image: danteev/texlive
+    steps:
+      - checkout
+      - run: 
+        command: |
+         cd pdf_build_src
+         sh build_pdf.sh
```
Comment (sappelhoff) [inline .circleci/config.yml]: you may want to add 

```
      - store_artifacts:
          path: rendered_pdf
```

see line 26 --> that would store a pdf in the "artifacts tab" (if the build succeeds) of the CircleCI output, see for example these files: https://circleci.com/gh/bids-standard/bids-specification/1799#artifacts/containers/0

Comment (Arshitha) [reply .circleci/config.yml]: Thank you! I have no prior experience with circleci so not really sure how to fix the following error: 
`Build-agent version 1.0.27691-0016e3a2 (2020-02-20T13:14:44+0000)

Configuration errors: 2 errors occurred:

* Error migrating config to version 2: 2 errors occurred:

* in job 'build_docs_pdf': step was not in the right format (unable to find step's name/details)
* in job 'build_docs_pdf': step was not in the right format (unable to find step's name/details)
* In step 2 definition: Step

  - command: cd pdf_build_src
sh build_pdf.sh

    run: map[]

has incorrect indentation,it should be formatted like:

  - step:
      option1: value
      option2: value`

I fixed the indentation but still getting the same error. Also, I'm not clear on why `build_docs` job is failing even though it was succeeding before adding `build_docs_pdf` job. I'd appreciate help on this one.

**Comment by sappelhoff** [comment]
@Arshitha I see you struggling with CircleCi and wanted to share this tool with you: https://support.circleci.com/hc/en-us/articles/360006735753-Validating-your-CircleCI-Configuration

perhaps it can be helpful to you: Instead of pushing, and waiting for remote CircleCI to run, you can debug locally :-)

**Comment by Arshitha** [comment]
I clearly am. I was trying to change the config on the circleci online dashboard but that wasn't helpful. Thanks for the resource! I wasn't happy about creating a new commit for each indentation fix.

File: .circleci/config.yml [hunk 10]
```diff
@@ -29,6 +29,19 @@ jobs:
...
           paths: site
 
+  build_docs_pdf:
+    working_directory: ~/bids-specification/pdf_build_src
+    docker:
+      - image: danteev/texlive:TL2017
+    steps:
+      - checkout:
+          path: ~/bids-specification
+      - run:
```
Comment (sappelhoff) [inline .circleci/config.yml]: I think you may be creating yourself some problems with the paths you are specifying in `working_directory` and `checkout`.

I would leave the `working_directory` line alone (i.e., remove line 33 here) ... then we will have the default working directory of `~/project` (see https://discuss.circleci.com/t/working-directory-is-no-longer-required-defaults-to-project/14363/2)

Then I would also remove line 38, corresponding to the `path` parameter of the `checkout` setting ... that will mean that the `bids-specification` will be cloned (checked out) to the default directory, which is the working directory, which will be `~/project`

Then you should take a look at your python scripts and how they handle paths. All the while assuming that you are working from a directory `~/project/`

**Comment by sappelhoff** [comment]
Cool, you got CircleCI to work :-) now there seems to be a path issue that prevents the build from actually producing the PDF:

```
README.md      build_pdf.sh  header.tex		 pandoc_script.py      src_copy
bids-spec.pdf  cover.tex     listings_setup.tex  process_markdowns.py
.  ..  images  src
Traceback (most recent call last):
  File "process_markdowns.py", line 185, in <module>
    extract_header_string()
  File "process_markdowns.py", line 67, in extract_header_string
    for i, line in enumerate(open(os.path.join(os.path.dirname(__file__), 'src_copy/CHANGES.md'))):
FileNotFoundError: [Errno 2] No such file or directory: 'src_copy/CHANGES.md'
[WARNING] Deprecated: markdown_github. Use gfm instead.
[WARNING] Could not fetch resource 'images/sub-0001_ses-001_acq-NAS_photo.jpg': replacing image with description
[WARNING] Could not fetch resource 'images/sub-0001_ses-001_acq-NAS_photo.jpg': replacing image with description
[WARNING] Could not fetch resource 'images/ieeg_electrodes1.png': replacing image with description
[WARNING] Could not fetch resource 'images/ieeg_electrodes2.png': replacing image with description
Error producing PDF.
! Unable to load picture or PDF file 'images/BIDS_logo.jpg'.
<to be read again> 
                   }
l.114 ...idth=0.6\textwidth]{images/BIDS_logo.jpg}

mv: cannot stat 'bids-spec.pdf': No such file or directory

```

I tried to help here: https://github.com/bids-standard/bids-specification/pull/400/files#r382491858

Let us know if you cannot solve this issue, so someone else can try to take over.

**Comment by Arshitha** [comment]
I was able to fix the path issue (for the most part) for circleci to build pdf docs successfully. However, I don't see it rendered under the artifacts tab. I'm looking more into this but if there's some quick fix that I'm missing here, I'd appreciate help. 
```
Uploading /root/bids-specification/pdf_build_src/rendered_pdf to rendered_pdf
  No artifact files found at /root/bids-specification/pdf_build_src/rendered_pdf
```
I'm pretty close to addressing the two issues you mentioned here: https://github.com/bids-standard/bids-specification/pull/400#pullrequestreview-361002053 so, I'd like to see it through. If I can't get it done in the next few hours, I'll let y'all know. 

I'm aware that debugging circleci errors took way too many commits but there was a learning curve to implementing it locally. So thought, pushing to github and checking circleci logs would be fastest? Correct if I'm wrong though. 

Also, before merging, would it be wise to create a new branch with cleaner commit history and just reference this (eventually, closed) PR for the value of discussion?

File: pdf_build_src/cover.tex (hunk 7 above)
[the remark this answers is not in the document]
Comment (Arshitha) [reply pdf_build_src/cover.tex]: Fixed this for both `header.tex` and `cover.tex`

File: pdf_build_src/header.tex [hunk 11]
```diff
@@ -0,0 +1,6 @@
+% header file
+\usepackage{fancyhdr}
+\pagestyle{fancy}
+\fancyhf{}
+\chead{Brain Imaging Data Structure v1.2.2 2020-02-12}
```
[the remark this answers is not in the document]
Comment (Arshitha) [reply pdf_build_src/header.tex]: see here: https://github.com/bids-standard/bids-specification/pull/400#discussion_r381209026

File: src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md (hunk 4 above)
Comment (Arshitha) [reply src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md]: So, I hadn't made any changes here. But in the commit before this, `src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md` was failing `TravisCI` tests. However, the fix mentioned here: https://github.com/bids-standard/bids-specification/blob/master/CONTRIBUTING.md worked. In that, `TravisCI` tests weren't failing anymore but I didn't really go into `src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md` to see what had done the trick. 

But just figured that this was caused by running `remark` on this file.

**Comment by Arshitha** [comment]
@sappelhoff 

I've addressed 


> the pdf_build_src/bids-spec.pdf PDF file is still part of this PR --> the file has to be removed before we can merge, because we do not want to git track each brand new PDF. Instead, we want CircleCI to build and host these PDFs, see: #400 (comment)

The circleci build was successful and the generated pdf is hosted under the artifacts tab. Also, made sure that the pdf version is removed from the repo post build. NOTE: as discussed here: https://github.com/bids-standard/bids-specification/pull/400#discussion_r381163494 the date in the header string is basically the build date. 

> I took the liberty to reformat your pdf_build_src/pandoc_script.py according to pep8 guidelines. As you can see e.g. with this online-pep8-checker, there were several formatting issues. Please add and commit this reformatted version (and also test it ... but it should work like before, I did not change its functionality, only the formatting)

After making final changes to the code, I ran it through the online checker. There are a few trailing spaces warnings but that's about it. 

However, while trying to get circleci to work, I essentially committed each time I wanted to debug. That has certainly made the commit history messy. Would it be better to create a new PR from a new branch with a cleaner commit history? and referencing this PR for the discussion value?

File: src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md (hunk 4 above)
Comment (Arshitha) [reply src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md]: Although, having angle brackets doesn't affect how its rendered on RTD or pdf. So for now, until further feedback, the most recent commit has the angle brackets.

File: .circleci/config.yml [hunk 12]
```diff
@@ -29,6 +29,22 @@ jobs:
...
+      - checkout:
+          path: ~/bids-specification
+      - run: 
+          name: generate pdf version docs 
+          command: sh build_pdf.sh
+      - store_artifacts:
+          path: bids-spec.pdf
+      - run:
+          name: remove pdf version from repo
+          command: rm bids-spec.pdf
```
Comment (sappelhoff) [inline .circleci/config.yml]: Checking the CircleCI build (e.g., https://circleci.com/gh/bids-standard/bids-specification/1880?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) I see that the pdf gets built and the artifact is uploaded ...

However, I do not see the "artifact tab" and thus cannot inspect the PDF. 

I wonder whether that's because of this last part: 

```
 - run:
          name: remove pdf version from repo
          command: rm bids-spec.pdf
```

Could you simply remove that? The pdf should not be committed to the repository anyhow, so the removal step should be unnecessary (and might be detrimental with regards to the artifact)

Comment (sappelhoff) [reply .circleci/config.yml]: EDIT: I just had to re-log into the CircleCI interface ... I do see the artifact now!

https://1880-150465237-gh.circle-artifacts.com/0/bids-spec.pdf

awesome job! :-)

**Comment by sappelhoff** [comment]
> That has certainly made the commit history messy. Would it be better to create a new PR from a new branch with a cleaner commit history? and referencing this PR for the discussion value?

Unless you know how to `rebase` and `squash` using `git`, it would indeed be best if you could make a new PR branched off the most recent `master` and by cherry picking the "good" commits of **this** PR.

of course, **IF** you know how to rebase this PR and squash the circle-ci-iteration-commits into a single commit. then you can just do it with this PR and we'll merge it.

File: .circleci/config.yml (hunk 12 above)
Comment (Arshitha) [reply .circleci/config.yml]: Great! Do you still think it's necessary to remove the quoted lines of code in the config file?

**Comment by Arshitha** [comment]
Sounds good! Will create a new PR with cleaner commit history (or try rebasing as well)

File: .circleci/config.yml (hunk 12 above)
Comment (sappelhoff) [reply .circleci/config.yml]: I think it doesn't matter. But since everything is working now, let's just leave it as is for now.

**Comment by effigies** [comment]
Hi just finally getting around to having a look at this, and it looks great!

Some notes:

1) Page 9 table. Is it possible to force the heading row to be a single line, even if that forces the table to be taller and descriptions overflow their box?
2) Page 18 table. Can we add a little bit of spacing between table rows? Or light (e.g., #cccccc) borders? The descriptions slam together in a way that isn't as readable as I could wish. This is true of most tables, but this is where it started to hit home for me.
3) Page 20 footnotes are not super-scripted or properly placed at the footer of the page.
4) Page 22 pre-formatted text ("Template"). Is it possible to adjust the rules for line breaks to prefer to go before/after underscores, after close square brackets or before open square brackets? That would be more natural.
5) Page 23 table. The "OPTIONAL" highlights some weirdness to the font. Particularly, `T` is out of alignment with the other caps. Any chance of changing fonts?
6) Page 26 table. This just looks bad. We should probably fix it in the main source, but just want to note how ugly it is in PDF.
7) Page 75. Hoo boy.
8) pp 76-77. Exponents are not super-scripted.
9) Page 82 table. The last entry is hard to read, due to the line break making `YYYY/MM` appear to be a new entry.

4 is likely a tar-pit, so it should not in any way hold things up. 7 you mention up top, so I will defer to whatever existing consensus there is.

**Comment by Arshitha** [comment]
@sappelhoff As discussed, here is a new PR with cleaner commit history: https://github.com/bids-standard/bids-specification/pull/427

**Comment by Arshitha** [comment]
@effigies Thanks for the feedback!

> Page 18 table. Can we add a little bit of spacing between table rows? Or light (e.g., #cccccc) borders? The descriptions slam together in a way that isn't as readable as I could wish. This is true of most tables, but this is where it started to hit home for me.

> Page 26 table. This just looks bad. We should probably fix it in the main source, but just want to note how ugly it is in PDF.

The issues with tables (2 & 6) were caused by running remark on one of the markdown files flagged by Travis CI but I've fixed them and the more recent pdf can found here: https://1927-150465237-gh.circle-artifacts.com/0/bids-spec.pdf as part of build from the newly (but almost identical) created PR here: https://github.com/bids-standard/bids-specification/pull/427  If I've missed something, I can work on it. 

Except for 1, 4 and 9,  I tried a few things to fix issues with superscripts and fonts but without much success. The consensus was to tackle them as part of a new enhancement issue in the future and I believe it'd be more efficient to deal with the others similarly as well.

**Comment by sappelhoff** [comment]
I agree with @Arshitha to tackle further issues in single iterative PRs and try to get this big one in first.

please continue discussion in #427
